### PR TITLE
Gitbook v2 Proposal

### PR DESCRIPTION
**Pull request by pwittrock**
Version 0.0.1 of *The Kubebuilder Book*.

Staged at http://book-staging.kubebuilder.io/

### CODE REVIEW

File: build/gitbook/cloudbuild.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  args: ["/usr/local/bin/gitbook", "build"]
+- name: "gcr.io/cloud-builders/docker"
+  dir: "docs/book"
+  args: ["build", ".", "-t", "gcr.io/kubebuilder/book"]
+- name: "gcr.io/cloud-builders/docker"
+  dir: "docs/book"
+  args: ["push", "gcr.io/kubebuilder/book"]
+- name: "gcr.io/cloud-builders/kubectl"
+  dir: "build/gitbook"
+  args: ["apply", "-f", "config/"]
```
[the remark this answers is not in the document]
Comment (pwittrock) [reply build/gitbook/cloudbuild.yaml]: I took this out.  It wasn't working well anyway :P

File: docs/book/basics/what_is_a_controller.md [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+Controllers must watch for events on the resource whose API they implement.  The ReplicaSet-controller
+watches for changes to ReplicaSets.  If a ReplicaSet is created, modified or deleted then the
+controller calls the Reconcile method with the key of the ReplicaSet.  Reconcile will read the ReplicaSet
+state and the state of all of its Pods.
+ 
+###### ReplicaSet Creation
+
+The following diagram shows a creation event triggering a reconcile.
+
+{% sequence %}
```
[the remark this answers is not in the document]
Comment (pwittrock) [reply docs/book/basics/what_is_a_controller.md]: :)

**Comment by droot** [comment]
@pwittrock Can we close this ? I guess we merged all of these in separate PR.

File: docs/book/basics/what_is_a_resource.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+```
+
+{% endmethod %}
+
+#### Labels, Selectors and Annotations
+
+Labels in ObjectMeta data are key-value pairs that may be queried to find objects matching a query.
+Labels are used to form logical connections between objects in a Kubernetes cluster.  For instance
+Services use labels to determine which Pods to direct traffic to, and Deployments use labels
+(along with OwnersReferences) to identify Pods they own.
```
[the remark this answers is not in the document]
Comment (janetkuo) [reply docs/book/basics/what_is_a_resource.md]: Yes, `OwnersReferences` is used by all workload controllers to address the overlapping label selectors problem.

**Comment by pwittrock** [comment]
Closing this for now since most of it was merged in another PR.

**Pull request closed without merging**